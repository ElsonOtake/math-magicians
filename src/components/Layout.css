nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 1rem 2rem;
}

ul {
  display: flex;
  gap: 1rem;
}

li {
  list-style: none;
}

h1 {
  font-size: 3rem;
  color: blue;
}

a {
  text-decoration: none;
}