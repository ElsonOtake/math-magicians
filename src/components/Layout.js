import React from 'react';
import { Link, Outlet } from 'react-router-dom';

const Layout = () => (
  <>
    <nav>
      <h1>Math Magicians</h1>
      <ul>
        <li>
          <Link to="/home">Home</Link>
        </li>
        <li>
          <Link to="/calculator">Calculator</Link>
        </li>
        <li>
          <Link to="/quote">Quote</Link>
        </li>
      </ul>
    </nav>

    <Outlet />
  </>
);

export default Layout;
