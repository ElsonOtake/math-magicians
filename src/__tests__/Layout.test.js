import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import Layout from '../components/Layout.js';

describe('Layout', () => {
  test('renders Layout component', () => {
    render(<Layout />, {wrapper: BrowserRouter});
    expect(screen.getByRole('navigation')).toBeInTheDocument();
    expect(screen.getByRole('navigation')).toBeVisible();
    expect(screen.getByRole('navigation')).not.toHaveClass();
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(screen.getByRole('heading')).toBeVisible();
    expect(screen.getByRole('heading')).not.toHaveClass();
    expect(screen.getByRole('heading')).toHaveTextContent('Math Magicians');
    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.getByRole('list')).toBeVisible();
    expect(screen.getByRole('list')).not.toHaveClass();
    expect(screen.getByText(/Math Magicians/)).toBeTruthy();
    expect(screen.getByText(/Home/)).toBeTruthy();
    expect(screen.getByText(/Calculator/)).toBeTruthy();
    expect(screen.getByText(/Quote/)).toBeTruthy();
    expect(screen.debug()).toMatchSnapshot();
  });
});