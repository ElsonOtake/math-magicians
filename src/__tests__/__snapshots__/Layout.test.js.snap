// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders Layout component 1`] = `undefined`;
